pkg/ccl/changefeedccl: BenchmarkEncoders failed

pkg/ccl/changefeedccl.BenchmarkEncoders [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20289400?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20289400?buildTab=artifacts#./artifacts/microbench/experiment) on master @ [ca0d4076c381609d6487040141c87e36a68b211f](https://github.com/cockroachdb/cockroach/commits/ca0d4076c381609d6487040141c87e36a68b211f):


```
BenchmarkEncoders/json/encodeValue/2cols/envelope=row/diff=false
BenchmarkEncoders/json/encodeValue/2cols/envelope=row/diff=false-32         	  261714	      4451 ns/op	    1210 B/op	      36 allocs/op
BenchmarkEncoders/json/encodeValue/2cols/envelope=row/diff=true
BenchmarkEncoders/json/encodeValue/2cols/envelope=row/diff=true-32          	  264046	      4420 ns/op	    1210 B/op	      36 allocs/op
BenchmarkEncoders/json/encodeValue/2cols/envelope=wrapped/diff=false
BenchmarkEncoders/json/encodeValue/2cols/envelope=wrapped/diff=false-32     	  220820	      5121 ns/op	    1571 B/op	      45 allocs/op
BenchmarkEncoders/json/encodeValue/2cols/envelope=wrapped/diff=true
BenchmarkEncoders/json/encodeValue/2cols/envelope=wrapped/diff=true-32      	  130863	      8595 ns/op	    2631 B/op	      75 allocs/op
    encoder_test.go:1020: column types: [refcursor int2 tuple{bpchar, time, float} int2 tuple{bpchar, time, float} time pg_lsn uuid bytes uuid tuple{bpchar, bpchar, refcursor} tsvector int bytes jsonb tuple{time, decimal, timestamp, float, int2vector}], keys: [refcursor int2 tuple{bpchar, time, float}]
BenchmarkEncoders/json/encodeKey/3cols
    encoder_test.go:982: unknown envelope type 
--- FAIL: BenchmarkEncoders/json/encodeKey/3cols
BenchmarkEncoders/json/encodeValue/3cols/envelope=row/diff=false
BenchmarkEncoders/json/encodeValue/3cols/envelope=row/diff=false-32         	   94213	     12077 ns/op	    3242 B/op	      98 allocs/op
BenchmarkEncoders/json/encodeValue/3cols/envelope=row/diff=true
BenchmarkEncoders/json/encodeValue/3cols/envelope=row/diff=true-32          	   93358	     12106 ns/op	    3242 B/op	      98 allocs/op
BenchmarkEncoders/json/encodeValue/3cols/envelope=wrapped/diff=false
BenchmarkEncoders/json/encodeValue/3cols/envelope=wrapped/diff=false-32     	   78459	     14331 ns/op	    3983 B/op	     121 allocs/op
BenchmarkEncoders/json/encodeValue/3cols/envelope=wrapped/diff=true
BenchmarkEncoders/json/encodeValue/3cols/envelope=wrapped/diff=true-32      	   46165	     25188 ns/op	    7042 B/op	     214 allocs/op
    encoder_test.go:1020: column types: [refcursor int2 tuple{bpchar, time, float} time int2 tuple{bpchar, time, float} time pg_lsn uuid bytes uuid tuple{bpchar, bpchar, refcursor} tsvector int bytes jsonb tuple{time, decimal, timestamp, float, int2vector} date tsvector bool int jsonb date tsquery pg_lsn time name tuple{bit, int2} string interval decimal tsvector], keys: [refcursor int2 tuple{bpchar, time, float} time]
BenchmarkEncoders/json/encodeKey/4cols
    encoder_test.go:982: unknown envelope type 
--- FAIL: BenchmarkEncoders/json/encodeKey/4cols
BenchmarkEncoders/json/encodeValue/4cols/envelope=row/diff=false
BenchmarkEncoders/json/encodeValue/4cols/envelope=row/diff=false-32         	   50930	     22671 ns/op	    5323 B/op	     162 allocs/op
BenchmarkEncoders/json/encodeValue/4cols/envelope=row/diff=true
BenchmarkEncoders/json/encodeValue/4cols/envelope=row/diff=true-32          	   50960	     21935 ns/op	    5323 B/op	     162 allocs/op
BenchmarkEncoders/json/encodeValue/4cols/envelope=wrapped/diff=false
BenchmarkEncoders/json/encodeValue/4cols/envelope=wrapped/diff=false-32     	   44356	     25238 ns/op	    6149 B/op	     186 allocs/op
BenchmarkEncoders/json/encodeValue/4cols/envelope=wrapped/diff=true
BenchmarkEncoders/json/encodeValue/4cols/envelope=wrapped/diff=true-32      	   27000	     42938 ns/op	   11355 B/op	     342 allocs/op
--- FAIL: BenchmarkEncoders/json
BenchmarkEncoders/csv
    encoder_test.go:1020: column types: [refcursor int2 tuple{bpchar, time, float} time], keys: [refcursor]
BenchmarkEncoders/csv/encodeValue/1cols/envelope=row/diff=false
BenchmarkEncoders/csv/encodeValue/1cols/envelope=row/diff=false-32          	  878407	      1178 ns/op	      13 B/op	       1 allocs/op
    encoder_test.go:1020: column types: [refcursor int2 int2 tuple{bpchar, time, float} time pg_lsn uuid bytes], keys: [refcursor int2]
BenchmarkEncoders/csv/encodeValue/2cols/envelope=row/diff=false
BenchmarkEncoders/csv/encodeValue/2cols/envelope=row/diff=false-32          	  526173	      2188 ns/op	      45 B/op	       5 allocs/op
    encoder_test.go:1020: column types: [refcursor int2 tuple{bpchar, time, float} int2 tuple{bpchar, time, float} time pg_lsn uuid bytes uuid tuple{bpchar, bpchar, refcursor} tsvector int bytes jsonb tuple{time, decimal, timestamp, float, int2vector}], keys: [refcursor int2 tuple{bpchar, time, float}]
BenchmarkEncoders/csv/encodeValue/3cols/envelope=row/diff=false
BenchmarkEncoders/csv/encodeValue/3cols/envelope=row/diff=false-32          	  126170	      9739 ns/op	     692 B/op	      25 allocs/op
    encoder_test.go:1020: column types: [refcursor int2 tuple{bpchar, time, float} time int2 tuple{bpchar, time, float} time pg_lsn uuid bytes uuid tuple{bpchar, bpchar, refcursor} tsvector int bytes jsonb tuple{time, decimal, timestamp, float, int2vector} date tsvector bool int jsonb date tsquery pg_lsn time name tuple{bit, int2} string interval decimal tsvector], keys: [refcursor int2 tuple{bpchar, time, float} time]
BenchmarkEncoders/csv/encodeValue/4cols/envelope=row/diff=false
BenchmarkEncoders/csv/encodeValue/4cols/envelope=row/diff=false-32          	   56528	     21029 ns/op	    1715 B/op	      66 allocs/op
--- FAIL: BenchmarkEncoders
FAIL

I250803 23:38:47.457420 1 (gostd) rand.go:268  [-] 1  random seed: -667270878070677090
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*BenchmarkEncoders.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53202

pkg/ccl/changefeedccl.BenchmarkEncoders [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20322595?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20322595?buildTab=artifacts#./artifacts/microbench/experiment) on master @ [166d6538fc5bc47b861d81c95e9dbfc19994a1da](https://github.com/cockroachdb/cockroach/commits/166d6538fc5bc47b861d81c95e9dbfc19994a1da):


```
BenchmarkEncoders/json/encodeValue/2cols/envelope=row/diff=false
BenchmarkEncoders/json/encodeValue/2cols/envelope=row/diff=false-32         	  177234	      6043 ns/op	    1571 B/op	      52 allocs/op
BenchmarkEncoders/json/encodeValue/2cols/envelope=row/diff=true
BenchmarkEncoders/json/encodeValue/2cols/envelope=row/diff=true-32          	  185032	      6007 ns/op	    1571 B/op	      52 allocs/op
BenchmarkEncoders/json/encodeValue/2cols/envelope=wrapped/diff=false
BenchmarkEncoders/json/encodeValue/2cols/envelope=wrapped/diff=false-32     	  161325	      6775 ns/op	    1946 B/op	      61 allocs/op
BenchmarkEncoders/json/encodeValue/2cols/envelope=wrapped/diff=true
BenchmarkEncoders/json/encodeValue/2cols/envelope=wrapped/diff=true-32      	   94639	     11874 ns/op	    3366 B/op	     107 allocs/op
    encoder_test.go:1020: column types: [timestamptz collatedstring{da_DK} int2vector collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} oidvector tuple{collatedstring{da_DK}, interval} date oidvector timetz float float tuple{timestamptz, uuid, decimal} bpchar int], keys: [timestamptz collatedstring{da_DK} int2vector]
BenchmarkEncoders/json/encodeKey/3cols
    encoder_test.go:982: unknown envelope type 
--- FAIL: BenchmarkEncoders/json/encodeKey/3cols
BenchmarkEncoders/json/encodeValue/3cols/envelope=row/diff=false
BenchmarkEncoders/json/encodeValue/3cols/envelope=row/diff=false-32         	   95269	     11710 ns/op	    3078 B/op	      96 allocs/op
BenchmarkEncoders/json/encodeValue/3cols/envelope=row/diff=true
BenchmarkEncoders/json/encodeValue/3cols/envelope=row/diff=true-32          	   94320	     11707 ns/op	    3078 B/op	      96 allocs/op
BenchmarkEncoders/json/encodeValue/3cols/envelope=wrapped/diff=false
BenchmarkEncoders/json/encodeValue/3cols/envelope=wrapped/diff=false-32     	   86227	     12861 ns/op	    3623 B/op	     111 allocs/op
BenchmarkEncoders/json/encodeValue/3cols/envelope=wrapped/diff=true
BenchmarkEncoders/json/encodeValue/3cols/envelope=wrapped/diff=true-32      	   49296	     22734 ns/op	    6518 B/op	     201 allocs/op
    encoder_test.go:1020: column types: [timestamptz collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} oidvector tuple{collatedstring{da_DK}, interval} date oidvector timetz float float tuple{timestamptz, uuid, decimal} bpchar int collatedstring{en_US} oid bytes float tuple{refcursor, name, bpchar} tuple{tuple{int2vector, string, float, tuple{bit}, string}, inet, vector, bpchar} bpchar tuple{timestamptz, "char"} timestamptz bit timestamptz timestamptz timestamptz int2 oidvector], keys: [timestamptz collatedstring{da_DK} int2vector tuple{timestamptz, oidvector}]
BenchmarkEncoders/json/encodeKey/4cols
    encoder_test.go:982: unknown envelope type 
--- FAIL: BenchmarkEncoders/json/encodeKey/4cols
BenchmarkEncoders/json/encodeValue/4cols/envelope=row/diff=false
BenchmarkEncoders/json/encodeValue/4cols/envelope=row/diff=false-32         	   14650	     81055 ns/op	   29307 B/op	     595 allocs/op
BenchmarkEncoders/json/encodeValue/4cols/envelope=row/diff=true
BenchmarkEncoders/json/encodeValue/4cols/envelope=row/diff=true-32          	   14617	     80443 ns/op	   29319 B/op	     596 allocs/op
BenchmarkEncoders/json/encodeValue/4cols/envelope=wrapped/diff=false
BenchmarkEncoders/json/encodeValue/4cols/envelope=wrapped/diff=false-32     	   14122	     83866 ns/op	   30252 B/op	     623 allocs/op
BenchmarkEncoders/json/encodeValue/4cols/envelope=wrapped/diff=true
BenchmarkEncoders/json/encodeValue/4cols/envelope=wrapped/diff=true-32      	    6763	    161019 ns/op	   59487 B/op	    1214 allocs/op
--- FAIL: BenchmarkEncoders/json
BenchmarkEncoders/csv
    encoder_test.go:1020: column types: [timestamptz collatedstring{da_DK} int2vector tuple{timestamptz, oidvector}], keys: [timestamptz]
BenchmarkEncoders/csv/encodeValue/1cols/envelope=row/diff=false
BenchmarkEncoders/csv/encodeValue/1cols/envelope=row/diff=false-32          	  474915	      2305 ns/op	      13 B/op	       2 allocs/op
    encoder_test.go:1020: column types: [timestamptz collatedstring{da_DK} collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} oidvector tuple{collatedstring{da_DK}, interval} date], keys: [timestamptz collatedstring{da_DK}]
BenchmarkEncoders/csv/encodeValue/2cols/envelope=row/diff=false
BenchmarkEncoders/csv/encodeValue/2cols/envelope=row/diff=false-32          	  186427	      6009 ns/op	     274 B/op	      11 allocs/op
    encoder_test.go:1020: column types: [timestamptz collatedstring{da_DK} int2vector collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} oidvector tuple{collatedstring{da_DK}, interval} date oidvector timetz float float tuple{timestamptz, uuid, decimal} bpchar int], keys: [timestamptz collatedstring{da_DK} int2vector]
BenchmarkEncoders/csv/encodeValue/3cols/envelope=row/diff=false
BenchmarkEncoders/csv/encodeValue/3cols/envelope=row/diff=false-32          	  102129	     11166 ns/op	     577 B/op	      24 allocs/op
    encoder_test.go:1020: column types: [timestamptz collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} oidvector tuple{collatedstring{da_DK}, interval} date oidvector timetz float float tuple{timestamptz, uuid, decimal} bpchar int collatedstring{en_US} oid bytes float tuple{refcursor, name, bpchar} tuple{tuple{int2vector, string, float, tuple{bit}, string}, inet, vector, bpchar} bpchar tuple{timestamptz, "char"} timestamptz bit timestamptz timestamptz timestamptz int2 oidvector], keys: [timestamptz collatedstring{da_DK} int2vector tuple{timestamptz, oidvector}]
BenchmarkEncoders/csv/encodeValue/4cols/envelope=row/diff=false
BenchmarkEncoders/csv/encodeValue/4cols/envelope=row/diff=false-32          	    9890	    104926 ns/op	   19066 B/op	     431 allocs/op
--- FAIL: BenchmarkEncoders
FAIL

I250810 11:28:00.785139 1 (gostd) rand.go:268  [-] 1  random seed: -1864905959756757965
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*BenchmarkEncoders.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/changefeedccl.BenchmarkEncoders [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20353368?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20353368?buildTab=artifacts#./artifacts/microbench/experiment) on master @ [8f5ae52a8f9a419efd0c4ab26464ff9b265ad1f3](https://github.com/cockroachdb/cockroach/commits/8f5ae52a8f9a419efd0c4ab26464ff9b265ad1f3):


```
BenchmarkEncoders/protobuf/encodeValue/1cols/envelope=bare/diff=false-32    	  305690	      3723 ns/op	    1107 B/op	      36 allocs/op
BenchmarkEncoders/protobuf/encodeValue/1cols/envelope=bare/diff=true
BenchmarkEncoders/protobuf/encodeValue/1cols/envelope=bare/diff=true-32     	  298592	      3730 ns/op	    1107 B/op	      36 allocs/op
BenchmarkEncoders/protobuf/encodeValue/1cols/envelope=enriched/diff=false
BenchmarkEncoders/protobuf/encodeValue/1cols/envelope=enriched/diff=false-32         	  322587	      3506 ns/op	    1121 B/op	      35 allocs/op
BenchmarkEncoders/protobuf/encodeValue/1cols/envelope=enriched/diff=true
BenchmarkEncoders/protobuf/encodeValue/1cols/envelope=enriched/diff=true-32          	  171471	      6518 ns/op	    2163 B/op	      69 allocs/op
    encoder_test.go:1029: column types: [timestamptz collatedstring{da_DK} collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} oidvector tuple{collatedstring{da_DK}, interval} date], keys: [timestamptz collatedstring{da_DK}]
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=wrapped/diff=false
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=wrapped/diff=false-32          	  126418	      8488 ns/op	    2688 B/op	      78 allocs/op
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=wrapped/diff=true
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=wrapped/diff=true-32           	   73647	     15109 ns/op	    4734 B/op	     143 allocs/op
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=bare/diff=false
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=bare/diff=false-32             	  152492	      7271 ns/op	    2101 B/op	      68 allocs/op
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=bare/diff=true
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=bare/diff=true-32              	  151801	      7264 ns/op	    2101 B/op	      68 allocs/op
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=enriched/diff=false
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=enriched/diff=false-32         	  156828	      7087 ns/op	    2115 B/op	      67 allocs/op
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=enriched/diff=true
BenchmarkEncoders/protobuf/encodeValue/2cols/envelope=enriched/diff=true-32          	   82484	     13590 ns/op	    4170 B/op	     132 allocs/op
    encoder_test.go:1029: column types: [timestamptz collatedstring{da_DK} int2vector collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} oidvector tuple{collatedstring{da_DK}, interval} date oidvector timetz float float tuple{timestamptz, uuid, decimal} bpchar int], keys: [timestamptz collatedstring{da_DK} int2vector]
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=wrapped/diff=false
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=wrapped/diff=false-32          	   69477	     16176 ns/op	    4577 B/op	     140 allocs/op
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=wrapped/diff=true
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=wrapped/diff=true-32           	   39824	     28624 ns/op	    8328 B/op	     259 allocs/op
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=bare/diff=false
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=bare/diff=false-32             	   77522	     14222 ns/op	    3819 B/op	     122 allocs/op
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=bare/diff=true
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=bare/diff=true-32              	   77828	     14395 ns/op	    3818 B/op	     122 allocs/op
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=enriched/diff=false
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=enriched/diff=false-32         	   78820	     14211 ns/op	    3831 B/op	     121 allocs/op
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=enriched/diff=true
BenchmarkEncoders/protobuf/encodeValue/3cols/envelope=enriched/diff=true-32          	   42571	     26695 ns/op	    7581 B/op	     240 allocs/op
    encoder_test.go:1029: column types: [timestamptz collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} collatedstring{da_DK} int2vector tuple{timestamptz, oidvector} oidvector tuple{collatedstring{da_DK}, interval} date oidvector timetz float float tuple{timestamptz, uuid, decimal} bpchar int collatedstring{en_US} oid bytes float tuple{refcursor, name, bpchar} tuple{tuple{int2vector, string, float, tuple{bit}, string}, inet, vector, bpchar} bpchar tuple{timestamptz, "char"} timestamptz bit timestamptz timestamptz timestamptz int2 oidvector], keys: [timestamptz collatedstring{da_DK} int2vector tuple{timestamptz, oidvector}]
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=wrapped/diff=false
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=wrapped/diff=false-32          	   13596	     87718 ns/op	   37038 B/op	     683 allocs/op
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=wrapped/diff=true
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=wrapped/diff=true-32           	    6826	    166900 ns/op	   72477 B/op	    1324 allocs/op
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=bare/diff=false
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=bare/diff=false-32             	   14158	     84443 ns/op	   35655 B/op	     645 allocs/op
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=bare/diff=true
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=bare/diff=true-32              	   14157	     83234 ns/op	   35649 B/op	     645 allocs/op
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=enriched/diff=false
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=enriched/diff=false-32         	   14181	     82699 ns/op	   35670 B/op	     644 allocs/op
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=enriched/diff=true
BenchmarkEncoders/protobuf/encodeValue/4cols/envelope=enriched/diff=true-32          	    6988	    164119 ns/op	   71221 B/op	    1288 allocs/op
--- FAIL: BenchmarkEncoders
FAIL

I250817 02:29:34.058631 1 (gostd) rand.go:268  [-] 1  random seed: 2913935753526421935
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*BenchmarkEncoders.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/encoder_test.go
+++ b/pkg/ccl/changefeedccl/encoder_test.go
@@ -1029,12 +1029,15 @@ func BenchmarkEncoders(b *testing.B) {
 				b.Logf("column types: %v, keys: %v", colTypes, colTypes[:numKeyCols])
 
 				if tc.benchEncodeKey {
-					b.Run(fmt.Sprintf("encodeKey/%dcols", numKeyCols),
-						func(b *testing.B) {
-							opts := changefeedbase.EncodingOptions{Format: tc.format}
-							bench(b, encodeKey, opts, updatedRows, prevRows)
-						},
-					)
+					testutils.RunValues(b, "envelope", tc.envelopes,
+						func(t *testing.B, envelope changefeedbase.EnvelopeType) {
+							b.Run(fmt.Sprintf("encodeKey/%dcols/envelope=%s", numKeyCols, envelope),
+								func(b *testing.B) {
+									opts := changefeedbase.EncodingOptions{Format: tc.format, Envelope: envelope}
+									bench(b, encodeKey, opts, updatedRows, prevRows)
+								},
+							)
+						})
 				}
 
 				for _, envelope := range tc.envelopes {

